src/primitives/decode.rs

```diff
@@ -750,19 +739,19 @@ impl std::error::Error for CharError {
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
 pub enum ChecksumError {
-    /// The checksum does not match the rest of the data.
-    InvalidChecksum,
+    /// The checksum residue is not valid for the data.
+    InvalidResidue,
     /// The checksum is not a valid length.
-    InvalidChecksumLength,
+    InvalidLength,
```

Comment:
In 8eb7954da33cbb9c5e620a049cff12cb1e1ff94b:

Here and in the display output, we should say "the checksummed string is not a valid length" or "the string is not a valid length for this checksum".

Reply:
I went with "the checksummed string is not a valid length". Thanks.